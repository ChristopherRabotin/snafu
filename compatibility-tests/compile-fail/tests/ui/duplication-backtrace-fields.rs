use snafu::{prelude::*, Backtrace};

#[derive(Debug, Snafu)]
enum EnumError {
    AVariant {
        // First backtrace, legitimate
        backtrace: Backtrace,

        // Trying to declare another field as backtrace, should be a duplicate
        #[snafu(backtrace)]
        my_backtrace: Backtrace,
    },
}

fn main() {}
