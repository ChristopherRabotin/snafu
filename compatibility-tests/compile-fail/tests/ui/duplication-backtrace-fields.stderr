error: Multiple `backtrace` attributes are not supported within an enum variant
  --> tests/ui/duplication-backtrace-fields.rs:10:17
   |
10 |         #[snafu(backtrace)]
   |                 ^^^^^^^^^
